### DISCUSSION TITLE
Keep @ARGV

### DISCUSSION THREAD

**Discussion by mche** [mojolicious/mojo#1687] [Ideas]
Hi, team.

Is it possible to patch scripts **morbo** & **hypnotoad** for one line:

`$ENV{MOJO_ARGV} = join " ", @ARGV;`

right before `getopt` in the [morbo](https://github.com/mojolicious/mojo/blob/master/script/morbo#L7) and [hypnotoad](https://github.com/mojolicious/mojo/blob/master/script/hypnotoad#L7)?

I would extend command line options on my apps.

Grand thanks. Regards.
♥ Mojolicious

**Comment by Grinnz** [comment]
morbo and hypnotoad as you linked are extremely simple scripts. I would recommend simply writing your own script that runs a Mojo::Server::Morbo or Mojo::Server::Hypnotoad as you desire. Just be aware that Mojo::Server::Hypnotoad will not see the new environment or arguments when hot-restarted (this is why it must be configured through app configuration).

**Reply by mche** [reply]
Ok, solved, thnk

**Reply by kraih** [reply]
Be aware that all environment variables in those scripts that are not mentioned specifically in the documentation are considered private, and will change in the future.

**Reply by kraih** [reply]
In fact i just made one such change. And the same will happen for Hypnotoad in the future. https://github.com/mojolicious/mojo/compare/f34c714f04f3...46af87f44e1d
